test/Microsoft.Azure.SignalR.AspNet.Tests/SignalRMessageParserTest.cs

```diff
@@ -0,0 +1,310 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.AspNet.SignalR;
+using Microsoft.AspNet.SignalR.Infrastructure;
+using Microsoft.AspNet.SignalR.Json;
+using Microsoft.AspNet.SignalR.Messaging;
+using Microsoft.Azure.SignalR.Protocol;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.Azure.SignalR.AspNet.Tests
+{
+    public class SignalRMessageParserTest
+    {
+        private readonly IDependencyResolver _resolver = GetDefaultResolver();
+        private readonly MemoryPool _pool = new MemoryPool();
+        private readonly JsonSerializer _serializer = new JsonSerializer();
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("key")]
+        [InlineData("pc-SignalRMessageParserTest")]
+        [InlineData("pcg-GroupSignalRMessageParserTest")]
+        [InlineData("c-GroupSignalRMessageParserTest")]
+        public void TestGetMessageWithUnknownKeyThrows(string key)
+        {
+            var hubs = new List<string> { };
+            var parser = new SignalRMessageParser(hubs, _resolver);
+            var raw = "<script type=\"\"></script>";
```

Comment:
Why a script tag?

Reply:
no special reason, to test various chars